pkg/providers/amifamily/ami.go

```diff
@@ -181,18 +214,19 @@ func (p *Provider) fetchAMIsFromSSM(ctx context.Context, ssmQuery string) (strin
 	return ami, nil
 }
 
-func (p *Provider) getAMIsFromSelector(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate) ([]AMI, error) {
-	ec2AMIs, err := p.fetchAMIsFromEC2(ctx, nodeTemplate.Spec.AMISelector)
+func (p *Provider) getAMIsFromSelector(ctx context.Context, selector map[string]string) ([]AMI, error) {
+	ec2AMIs, err := p.fetchAMIsFromEC2(ctx, selector)
 	if err != nil {
 		return nil, err
 	}
-	if len(ec2AMIs) == 0 {
-		return nil, fmt.Errorf("no amis exist given constraints")
-	}
 	var amis []AMI
 	for _, ec2AMI := range ec2AMIs {
 		amis = append(amis, AMI{*ec2AMI.Name, *ec2AMI.ImageId, *ec2AMI.CreationDate, p.getRequirementsFromImage(ec2AMI)})
 	}
+	if len(ec2AMIs) == 0 {
+		logging.FromContext(ctx).Errorf("no amis exist given constraints")
```

Comment:
Can you move this back to where it was originally? I don't think it needs to be moved, and keeping it as it was before is a good way to make sure things work as expected.